gstd/src/msg/async.rs

```diff
@@ -51,15 +52,17 @@ where
             "Somebody created a future with the MessageId that never ended in static replies!"
         ),
         ReplyPoll::Pending => Poll::Pending,
-        ReplyPoll::Some((actual_reply, reply_code)) => {
+        ReplyPoll::Some((payload, reply_code)) => {
             // Remove lock after waking.
             async_runtime::locks().remove(msg_id, waiting_reply_to);
 
-            if !reply_code.is_success() {
-                return Poll::Ready(Err(Error::ReplyCode(reply_code)));
+            match reply_code {
+                ReplyCode::Success(_) => Poll::Ready(f(payload)),
+                ReplyCode::Error(reason) => {
+                    Poll::Ready(Err(Error::ErrorReply(payload.into(), reason)))
+                }
+                ReplyCode::Unsupported => Poll::Ready(Err(Error::UnsupportedReply(payload))),
```

Comment:
I actually still have no idea how to deal with this error in case of new reply code variant to not break older program. Maybe we need to discuss it with our superiors? I want to ask you to add clear module documentation about this behavior at least